### ISSUE TITLE
Tiles are getting attached via ID rather than filename

### ISSUE THREAD

**Issue by cs96and** [KayelGee/token-attacher#17]
* Foundry 0.7.9
* Token Attacher 3.2.5
* Dnd5e 1.2.4
* All other modules disabled

I'm trying to attach a tile to a token and set it as the prototype token.  Most of the time, the tile seems to be attached via a temporary ID, rather than the filename.  That means that once the prototype is set, and I drag a second token onto the map, they both refer to the same tile.  Dragging either token results in the same tile moving.

If the tile is deleted, then the ID is no longer valid.  Dragging a new token onto the map results in the tile not being displayed.

I have got it to work correctly a couple of times, but I'm not sure what I did differently to make it work.

Here's a snippet from the exported JSON...

```json
  "token": {
    "flags": {
      "token-attacher": {
        "attached": {
          "Tile": [
            "WriNpiMk0wL89qDm"
          ]
        },
        "pos": {
          "xy": {
            "x": 0,
            "y": 0
          },
          "center": {
            "x": 400,
            "y": 400
          }
        }
      }
    },
```

**Comment by KayelGee** [comment]
The id get's replaced by the current id after the elements have been dropped on the canvas. It should always reference the new element.
If you're dragging elements to the canvas too quickly or some other update is happening in the background it can happen that the attachements don't get updated correctly.
This shouldn't happen regularly.
As this report is for 3.2.5 I suggest you try out 4.0. There have been a lot of code changes so maybe I might've optimized that part better so it doesn't happen so often anymore I'm not sure right now. At least there's a message that says when to token is ready to use.

If it still happens regularly for you in 4.0 then open a new ticket. I'll close this one.

**Comment by cs96and** [comment]
All seems to be fixed after updating to 4.0, thanks!
